unicorn-core/src/test/scala/org/virtuslab/unicorn/repositories/UsersRepositoryTest.scala

```diff
@@ -59,124 +60,143 @@ trait UsersRepositoryTest extends OptionValues {
-
-      UsersRepository.allIds() shouldEqual ids
+  it should "query all ids" in runWithRollback {
+    val users = Seq(
+      UserRow(None, "test1@email.com", "Krzysztof", "Nowak"),
+      UserRow(None, "test2@email.com", "Janek", "Nowak"),
+      UserRow(None, "test3@email.com", "Marcin", "Nowak")
+    )
+
+    val actions = for {
+      _ <- UsersRepository.create
+      ids <- UsersRepository saveAll users
+      allIds <- UsersRepository.allIds()
+    } yield (ids, allIds)
+
+    actions map {
+      case (ids, allIds) =>
+        allIds shouldEqual ids
+    }
   }
 
-  it should "sort users by id" in rollback {
-    implicit session =>
-      // setup
-      UsersRepository.create()
-
-      val users = Seq(
-        UserRow(None, "test1@email.com", "Krzysztof", "Nowak"),
-        UserRow(None, "test2@email.com", "Janek", "Nowak"),
-        UserRow(None, "test3@email.com", "Marcin", "Nowak")
-      )
-
-      val ids = UsersRepository saveAll users
-      val usersWithIds = (users zip ids).map { case (user, id) => user.copy(id = Some(id)) }
-
-      UsersRepository.findAll().sortBy(_.id) shouldEqual usersWithIds
+  it should "sort users by id" in runWithRollback {
+    val users = Seq(
+      UserRow(None, "test1@email.com", "Krzysztof", "Nowak"),
+      UserRow(None, "test2@email.com", "Janek", "Nowak"),
+      UserRow(None, "test3@email.com", "Marcin", "Nowak")
+    )
+
+    val actions = for {
+      _ <- UsersRepository.create
+      ids <- UsersRepository saveAll users
+      users <- UsersRepository.findAll()
+    } yield (ids, users)
+
+    actions map {
+      case (ids, users) =>
+        val usersWithIds = (users zip ids).map { case (user, id) => user.copy(id = Some(id)) }
+        users.sortBy(_.id) shouldEqual usersWithIds
+    }
   }
 
-  it should "query multiple users by ids" in rollback {
-    implicit session =>
-      // setup
-      UsersRepository.create()
-
-      val users = Seq(
-        UserRow(None, "test1@email.com", "Krzysztof", "Nowak"),
-        UserRow(None, "test2@email.com", "Janek", "Nowak"),
-        UserRow(None, "test3@email.com", "Marcin", "Nowak")
-      )
-
-      val ids = UsersRepository saveAll users
-      val usersWithIds = (users zip ids).map { case (user, id) => user.copy(id = Some(id)) }
-      UsersRepository.findAll().size shouldEqual 3
-
-      val selectedUsers = Seq(usersWithIds.head, usersWithIds.last)
-
-      UsersRepository.findByIds(selectedUsers.flatMap(_.id)) shouldEqual selectedUsers
+  it should "query multiple users by ids" in runWithRollback {
+    val users = Seq(
+      UserRow(None, "test1@email.com", "Krzysztof", "Nowak"),
+      UserRow(None, "test2@email.com", "Janek", "Nowak"),
+      UserRow(None, "test3@email.com", "Marcin", "Nowak")
+    )
+
+    val actions = for {
+      _ <- UsersRepository.create
+      ids <- UsersRepository saveAll users
+      allUsers <- UsersRepository.findAll
+      selectedUsers <- DBIO.sequence({
```

Comment:
`DBIO` is not needed.

    selectedUsers: Seq[UserRow] = {
        val usersWithIds = (users zip ids).map { case (user, id) => user.copy(id = Some(id)) }
        Seq(usersWithIds.head, usersWithIds.last)
    }